### ISSUE TITLE
Passing a Reason header for a CANCEL method through a sems

### ISSUE THREAD

**Issue by denyspozniak** [sems-server/sems#164]
Hello!

We have faced a problem of passing a Reason header for a CANCEL method through a sems.
Is there any way to fix this?

![image](https://user-images.githubusercontent.com/22612345/99302695-644e0580-2858-11eb-84ed-077cbac63068.png)

**Comment by mkezys** [comment]
Hi,

CANCEL is so-called hop-by-hop message. E.g. your first CANCEL is consumed by SEMS and the second one is generated by SEMS. SEMS is not re-sending your CANCEL. More details: https://thanhloi2603.wordpress.com/2017/05/28/introduction-to-sip-bye-cancel-and-hop-by-hop-messages/

**Comment by denyspozniak** [comment]
Thanks!
Yes, I understand that, but the task is to save this field when "forwarding" a message. But there are a number of cases where it is necessary.
For example, depending on the Reason for the canceled incoming call, a callee device may report it as a missed call (if the Reason header indicates a caller cancelling) or not (if the Reason header indicates that the call has established somewhere else, due to parallel forking).

**Comment by mkezys** [comment]
Yes, that would be great to pass this. SEMS identifies it as 'hdrs'

[handle_sip_request, SipCtrlInterface.cpp:782] DEBUG:  hdrs = <Reason: SIP;cause=200;text="Call completed elsewhere"..>

A quick look at the code shows like it should be copied to the generated CANCEL:

https://github.com/sems-server/sems/blob/master/core/sip/trans_layer.cpp#L1509

**Comment by ThomasSevestre** [comment]
Do you use SBC module ?

**Comment by denyspozniak** [comment]
@ThomasSevestre yes
`application = sbc`

**Comment by ThomasSevestre** [comment]
I have done most of the work but it is not mergeable as is. If you can wait I'll make a PR but I'm not sure when I'll be able to.

**Comment by denyspozniak** [comment]
Thanks!
Of course, I'll wait.

**Comment by hecko** [comment]
This has been fixed on `master` (commit fecacc9). The SBC CallLeg now forwards the RFC 3326 `Reason:` header from an incoming CANCEL onto the peer leg's BYE. Other CANCEL headers are filtered out to avoid leaking hop-by-hop metadata.
